raiden/tests/integration/test_settlement.py

```diff
@@ -126,6 +130,101 @@ def test_settle_is_automatically_called(raiden_network, token_addresses, deposit
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        views.state_from_app(alice_app),
+        alice_app.raiden.default_registry.address,
+        token_address,
+    )
+
+    token_network = views.get_token_network_by_identifier(
+        views.state_from_app(alice_app),
+        token_network_identifier,
+    )
+
+    channel_state = get_channelstate(alice_app, bob_app, token_network_identifier)
+    channel_identifier = channel_state.identifier
+
+    assert channel_identifier in token_network.partneraddresses_to_channels[
+        bob_app.raiden.address
+    ]
+
+    alice_to_bob_amount = 10
+    identifier = 1
+    transfer_1_secret = pending_mediated_transfer(
+        raiden_network,
+        token_network_identifier,
+        alice_to_bob_amount,
+        identifier,
+    )
+    transfer_1_secrethash = sha3(transfer_1_secret)
+
+    alice_bob_channel_state = get_channelstate(alice_app, bob_app, token_network_identifier)
+    lock = channel.get_lock(alice_bob_channel_state.our_state, transfer_1_secrethash)
+
+    # This is the current state of the protocol:
+    #
+    #    A -> B LockedTransfer
+    #    B -> A SecretRequest
+    #    - protocol didn't continue
+    assert_synced_channel_state(
+        token_network_identifier,
+        alice_app, deposit, [lock],
+        bob_app, deposit, [],
+    )
+
+    # Verify lock is registered in both channel states
+    alice_channel_state = get_channelstate(alice_app, bob_app, token_network_identifier)
+    assert transfer_1_secrethash in alice_channel_state.our_state.secrethashes_to_lockedlocks
+
+    bob_channel_state = get_channelstate(bob_app, alice_app, token_network_identifier)
+    assert transfer_1_secrethash in bob_channel_state.partner_state.secrethashes_to_lockedlocks
+
+    alice_chain_state = views.state_from_raiden(alice_app.raiden)
+    assert transfer_1_secrethash in alice_chain_state.payment_mapping.secrethashes_to_task
+
+    # Wait for the expiration to trigger with some additional buffer
+    # time for processing (+2) blocks.
+    waiting.wait_for_block(
+        alice_app.raiden,
+        lock.expiration + (DEFAULT_NUMBER_OF_CONFIRMATIONS_BLOCK + 2),
+        DEFAULT_RETRY_TIMEOUT,
+    )
+
+    alice_channel_state = get_channelstate(alice_app, bob_app, token_network_identifier)
+    assert transfer_1_secrethash not in alice_channel_state.our_state.secrethashes_to_lockedlocks
+
+    # Verify bob received the message and processed the LockExpired message
+    bob_channel_state = get_channelstate(bob_app, alice_app, token_network_identifier)
+    assert transfer_1_secrethash not in bob_channel_state.partner_state.secrethashes_to_lockedlocks
+
+    alice_chain_state = views.state_from_raiden(alice_app.raiden)
+    assert transfer_1_secrethash not in alice_chain_state.payment_mapping.secrethashes_to_task
+
+    # Make another transfer
+    alice_to_bob_amount = 10
+    identifier = 2
+    transfer_2_secret = pending_mediated_transfer(
```

Comment:
this should have caught this bug: https://github.com/raiden-network/raiden/pull/2216#discussion_r213037207

With a unit test you can assert on the result of the function `is_valid_lockexpired`